README.md

```diff
@@ -0,0 +1,280 @@
+values and add missing values.  Using:
+```yaml
+kernel_settings_bootloader_cmdline:
+  - name: spectre_v2
+    value: "off"
+  - name: nopti
+  - name: quiet
+  - name: splash
+```
+will result in the values `spectre_v2=off nopti quiet splash` being added to,
+or replacing, the existing `bootloader` `cmdline` values.  If you want to
+remove whatever was there, and replace it with your specified values, use
+`previous: replaced` in the value list:
+```yaml
+kernel_settings_bootloader_cmdline:
+  - previous: replaced
+  - name: spectre_v2
+    value: "off"
+  - name: nopti
+  - name: quiet
+  - name: splash
+```
+then the values `spectre_v2=off nopti quiet splash` will *replace* the
+existing `cmdline` values, if any.  If you want to remove specific values, use
+`state: absent` on the settings you want to remove:
+```yaml
+kernel_settings_bootloader_cmdline:
+  - name: spectre_v2
+    state: absent
+  - name: nopti
+    state: absent
+```
+if the previous value was `spectre_v2=off nopti quiet splash`, the items
+`spectre_v2=off` and `nopti` will be removed from the list, and the final
+value will be `quiet splash`.
+
+## Dependencies
+
+The `tuned` package is required for the default provider.
+
+## Example Playbook
+
+```yaml
+- hosts: all
+  vars:
+    kernel_settings_sysctl:
+      - name: fs.epoll.max_user_watches
+        value: 785592
+      - name: fs.file-max
+        value: 379724
+      - name: kernel.threads-max
+        state: absent
+    kernel_settings_sysfs:
+      - name: /sys/kernel/debug/x86/pti_enabled
+        value: 0
+      - name: /sys/kernel/debug/x86/retp_enabled
+        value: 0
+      - name: /sys/kernel/debug/x86/ibrs_enabled
+        value: 0
+    kernel_settings_transparent_hugepages: madvise
+    kernel_settings_transparent_hugepages_defrag: defer
+    kernel_settings_bootloader_cmdline:
+      - name: spectre_v2
+        value: "off"
+      - name: nopti
+      - name: quiet
+      - name: splash
+  roles:
+    - linux-system-roles.kernel_settings
+```
+
+## Warnings
+
+The `kernel_settings` role will cause other `sysctl` settings to be applied
+when using the `tuned` implementation, which is the default. For example, if
+you manually edit `/etc/sysctl.d/` files, or if the `sysctl.d` files are
+installed by some system package.  On Fedora, installing the `libreswan`
+package provides `/etc/sysctl.d/50-libreswan.conf`.  Using the
+`kernel_settings` role will cause this file to be reloaded and reapplied.  If
+this behavior is not desired, you will need to edit the `tuned` configuration
+on the managed hosts in `/etc/tuned/tuned-main.conf` and set
+`reapply_sysctl=0`.
+
+The settings you apply with the `kernel_settings` role may conflict with other
+settings.  For example, if you manually run the `sysctl` command, or manually
+edit `/etc/sysctl.d/` files, or if the `sysctl.d` files are installed by some
+system package, they may set the same values you are setting with the
+`kernel_settings` role.
```

Comment:
@richm Can you please document which one takes precedence in case of conflict - the `sysctl.d` files or the settings done by the role? Or is it intentionally left unspecified?

Reply:
> @richm Can you please document which one takes precedence in case of conflict - the `sysctl.d` files or the settings done by the role? Or is it intentionally left unspecified?

https://github.com/linux-system-roles/kernel_settings/pull/5